### PR TITLE
Target Android 11 (DEV)

### PR DESCRIPTION
**Pull request by mtwalli**
- To be tested after https://github.com/corona-warn-app/cwa-app-android/pull/3272
https://github.com/corona-warn-app/cwa-app-android/pull/3272  upgraded mockK which was one of the reasons to back-off last time.

TODO
- [x] Screenshots tests are not affected by this change

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -484,6 +484,10 @@ workflows:
           requires:
             - ktlint_check
             - detekt
+      - firebase_screenshots:
+          requires:
+            - ktlint_check
+            - detekt
```
Comment (mtwalli) [inline .circleci/config.yml:490]: Revert

File: Corona-Warn-App/build.gradle [hunk 2]
```diff
@@ -22,12 +22,12 @@ android {
...
     println("Current VERSION_BUILD: ${VERSION_BUILD}")
 
-    compileSdkVersion 29
+    compileSdkVersion 30
     buildToolsVersion "30.0.3"
     defaultConfig {
         applicationId 'de.rki.coronawarnapp'
         minSdkVersion 23
-        targetSdkVersion 29
+        targetSdkVersion 30
```
Comment (mtwalli) [inline Corona-Warn-App/build.gradle:30]: I think internal pipelines support API 30 , @kaddaSz  is that right?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/DebugLogFragment.kt [hunk 3]
```diff
@@ -174,7 +174,7 @@ class DebugLogFragment : Fragment(R.layout.bugreporting_debuglog_fragment), Auto
         MaterialAlertDialogBuilder(requireContext()).apply {
             setTitle(R.string.errors_generic_headline_short)
             setMessage(R.string.debugging_debuglog_start_low_storage_error)
-            setPositiveButton(android.R.string.yes) { _, _ -> /* dismiss */ }
+            setPositiveButton(android.R.string.ok) { _, _ -> /* dismiss */ }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/debuglog/ui/DebugLogFragment.kt:177]: finally right name 😀

File: Corona-Warn-App/build.gradle [hunk 4]
```diff
@@ -314,7 +314,7 @@ dependencies {
     // ANDROID STANDARD
     def nav_version = "2.3.3"
     implementation 'androidx.appcompat:appcompat:1.2.0'
-    implementation 'androidx.core:core-ktx:1.3.2'
+    implementation 'androidx.core:core-ktx:1.5.0'
```
Comment (mtwalli) [inline Corona-Warn-App/build.gradle:317]: To use `WindowInsetsControllerCompat` which has a nice system UI visibility handling

File: Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/datadonation/ui/DataDonationTestFragment.kt [hunk 5]
```diff
@@ -55,20 +55,20 @@ class DataDonationTestFragment : Fragment(R.layout.fragment_test_datadonation),
...
+            val intent = ShareCompat.IntentBuilder(requireActivity())
+                .setType("text/plain")
+                .setSubject("JWS")
+                .setText(jws)
+                .createChooserIntent()
             startActivity(intent)
         }
 
         vm.copyAnalyticsEvent.observe2(this) { analytics ->
-            val intent = ShareCompat.IntentBuilder.from(requireActivity()).apply {
```
Comment (mtwalli) [inline Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/datadonation/ui/DataDonationTestFragment.kt:67]: Deprecated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenFragment.kt [hunk 6]
```diff
@@ -1,19 +1,12 @@
...
 import android.view.View
-import android.view.View.SYSTEM_UI_FLAG_FULLSCREEN
-import android.view.View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
-import android.view.View.SYSTEM_UI_FLAG_IMMERSIVE
-import android.view.View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
-import android.view.View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
-import android.view.View.SYSTEM_UI_FLAG_LAYOUT_STABLE
-import android.view.View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
-import android.view.View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
-import android.view.View.SYSTEM_UI_FLAG_VISIBLE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/qrcode/fullscreen/QrCodeFullScreenFragment.kt:15]: System UI flags are deprecated in Android 11

**Review by kaddaSz — DISMISSED** [review]
Fine with me...

**Review by d4rken — APPROVED** [review]
Code changes lgtm (did not test on device)

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**